ItemLevelDownloader/Engine/worker.ts

```diff
@@ -0,0 +1,48 @@
+import { Logger } from './logger';
+export class Worker<T> {
+    private executionFunc: (item : T) => Promise<void>;
+    private getNextItem: () => T;
+    private canExit: () => boolean;
+
+    constructor(func: (item : T) => Promise<void>, getNextItem: () => T, canExit: () => boolean) {
+        this.executionFunc = func;
+        this.getNextItem = getNextItem;
+        this.canExit = canExit;
+    }
+
+    init(): Promise<void> {
```

Comment:
have an id for worker and log it in messages to diagnose issues involving multiple workers?